zilliqa/src/contracts/deposit.sol

```diff
@@ -138,123 +366,173 @@ contract Deposit {
         require(peerId.length == 38);
         require(signature.length == 96);
 
-        require(_stakerKeys.length < _maximumStakers, "too many stakers");
-
         // Verify signature as a proof-of-possession of the private key.
         bool pop = _popVerify(blsPubKey, signature);
         require(pop, "rogue key check");
 
-        uint256 keyIndex = _stakersMap[blsPubKey].keyIndex;
-        if (keyIndex == 0) {
-            // The staker will be at index `_stakerKeys.length`. We also need to add 1 to avoid the 0 sentinel value.
-            _stakersMap[blsPubKey].keyIndex = _stakerKeys.length + 1;
-            _stakerKeys.push(blsPubKey);
+        Staker storage staker = _stakersMap[blsPubKey];
+        // This must be a new staker, meaning the control address must be zero.
+        require(staker.controlAddress == address(0), "staker already exists");
```

Comment:
The name of this fn is misleading. Since a staker can call this successfully only once is something like `initialiseStaker()` more clear

Reply:
@DrZoltanFazekas Any opinions on naming preference?